src/goto-symex/execution_state.cpp

```diff
@@ -715,15 +732,6 @@ void execution_statet::execute_guard()
   if (active_thread != last_active_thread)
     target->assumption(
       guardt().as_expr(), parent_guard, get_active_state().source, first_loop);
-  // Check to see whether or not the state guard is false, indicating we've
-  // found an unviable interleaving. However don't do this if we didn't
-  // /actually/ switch between threads, because it's acceptable to have a
-  // context switch point in a branch where the guard is false (it just isn't
-  // acceptable to permit switching).
-  if (
-    last_active_thread != active_thread &&
-    is_cur_state_guard_false(threads_state[active_thread].guard.as_expr()))
-    interleaving_unviable = true;
```

Comment:
It is too early to set interleaving_unviable because the active_thread guard will be updated later in merge_gotos(), Instead set the value after merge_gotos()